### ISSUE TITLE
[BUG] TopDownReconciliator changes top level TimeSeries

### ISSUE THREAD

**Issue by DavidKleindienst** [unit8co/darts#1582]
**Describe the bug**
Using `TopDownReconciliator.fit_transform` changes the top level `TimeSeries`.
This should not happen, as reconcilliation should be performed by interpreting the forecasts of the bottom level as proportions of the top level, leaving the top level unchanged (see https://otexts.com/fpp3/single-level.html).

**To Reproduce**
```python
import pandas as pd
import numpy as np
from darts import TimeSeries, concatenate
from darts.dataprocessing.transformers import TopDownReconciliator

dates = pd.date_range("2020-01-01", "2020-12-31", freq = "D")
nr_dates = len(dates)
t1=TimeSeries.from_times_and_values(dates, 3*np.ones(nr_dates), columns=["T1"])
t2=TimeSeries.from_times_and_values(dates, 5*np.ones(nr_dates), columns=["T2"])
t3=TimeSeries.from_times_and_values(dates, np.ones(nr_dates), columns=["T3"])
tsum=TimeSeries.from_times_and_values(dates, 9*np.ones(nr_dates), columns=["T_sum"])

ts = concatenate([t1,t2,t3,tsum], axis="component")
hierarchy = {"T1": ["T_sum"], "T2": ["T_sum"], "T3": ["T_sum"]}
ts = ts.with_hierarchy(hierarchy)
ts_reconc = TopDownReconciliator().fit_transform(ts)

ts_reconc["T_sum"] == ts["T_sum"]
```
-> returns `False`

**Expected behavior**
Should return `True`, because TopDown reconciliation should not alter the top level

**System (please complete the following information):**
 - Python version: 3.10.8
 - darts version: 0.23.1

**Comment by DavidKleindienst** [comment]
Looking a bit more into this, the problem seems to be that the Reconcilliator depends on the order of components and interprets the first one as top level.
In the example above, replacing 
`ts = concatenate([t1,t2,t3,tsum], axis="component")`
with
`ts = concatenate([tsum,t1,t2,t3], axis="component")`
leads to the expected result

**Comment by DavidKleindienst** [comment]
The `BottomUpReconciliator` has exactly the same problem.
The problem is, that for both of them, `get_projection_matrix` assumes that the components of a `TimeSeries` are ordered from top to bottom. I think this is clearly a bug, as this prerequisite is not documented and also really unintuitive, given that it's possible to freely specify the hierarchy.

**Comment by hrzn** [comment]
Hi @DavidKleindienst , thanks for reporting. Indeed that looks like a bug, the behavior should definitely not be dependent on the orders of the components in the series! We should take a look at this asap (let us know if you want to help).
